libusb/driver.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libusb/driver.c
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <usbdriver.h>
+#include <sys/msg.h>
+#include <string.h>
+#include <unistd.h>
+
+
+static struct {
+	unsigned port;
+} usbdrv_common;
+
+
+int usb_connect(const usb_device_id_t *filters, int nfilters, unsigned drvport)
+{
+	msg_t msg = { 0 };
+	usb_msg_t *umsg = (usb_msg_t *)&msg.i.raw;
+	oid_t oid;
+
+	while (lookup("/dev/usb", NULL, &oid) < 0)
+		usleep(1000000);
+
+	msg.type = mtDevCtl;
+	msg.i.size = sizeof(usb_device_id_t) * nfilters;
```

Comment:
Somewhat more roboust way of using sizeof
```suggestion
	msg.i.size = sizeof(*filters) * nfilters;
```